### ISSUE TITLE
Agent not honoring DD_COLLECT_EC2_TAGS when ec2 integration is turned on

### ISSUE THREAD

**Issue by JohnlNguyen** [DataDog/datadog-agent#4092]
**Additional environment details (Operating System, Cloud provider, etc):**
datadog/agent:6.11.3
datadog/cluster-agent:1.3.1

**Steps to reproduce the issue:**
1. Set DD_COLLECT_EC2_TAGS: false
2. Turn on ec2 integration https://docs.datadoghq.com/integrations/amazon_ec2/
3. dd-agent still collects ec2 tags in metrics

**Describe the results you received:**
Metrics have ec2 tags
**Describe the results you expected:**
Metrics should not have ec2 tags

**Additional information you deem important (e.g. issue happens only occasionally):**

DD agent should honor the DD_COLLECT_EC2_TAGS rather than requiring users to modify the IAM policy to disable `ec2:DescribeInstances` in order to stop collecting ec2 tags

**Comment by semoac** [comment]
Same problem. We are trying to use dogstatsd but the custom metrics count just exploded because of this.

**Comment by 2rs2ts** [comment]
I had a support ticket open about our events having a bunch of garbage tags for the past few months. Forgot about this known issue. I have linked support to it. Thank you @semoac for commenting, this might help get a solution actually rolling.

**Comment by grosser** [comment]
@JohnlNguyen can you elaborate on where you did the "modify the IAM policy to disable ec2:DescribeInstances" I'm trying to do the same to fix the same issue 😞

**Comment by stefansedich** [comment]
Is there any update from anyone at DataDog on this? I have tried to cover this with support a number of times but have not had any luck getting a solution.

**Comment by 2rs2ts** [comment]
> Is there any update from anyone at DataDog on this?

Sort of. I apologize for not circling back here to comment on the issue about what a DataDog lead solutions engineer told me in Jan 2021 before closing out my ticket.

> I'm reaching out to follow up on this ticket thread, after a period of inactivity. Firstly, I'd like to apologize for the silence on our end here. After performing an audit on our side and reviewing your case in more detail, I can see that this bug / issue has been triaged by our product and engineering teams to be worked on, but I don't yet have a firm ETA for when this will be completed.

So, it's in their backlog but not roadmapped, most likely. I won't share the name of that lead solutions engineer publicly for their privacy's sake. What I will do is make a follow-up ticket saying "hey, don't forget about us!" Maybe the fact that my past self got far enough to get a lead solutions engineer involved will have more "weight" with them and they'll bump up the priority on fixing it... but IME every time I report a bug, it takes them years to fix it, if they fix it at all. Of course, I seem to only report the really tricky bugs that take sometimes _weeks_ of discussion to get them to nail it down, so, take that with a grain of salt. I'm not trying to shame them too much, just a tiny bit  🤏 

I'm bad at remembering to follow up with GitHub threads that aren't active and showing up in my notifications a lot, so don't expect a response from me unless you ping me in a couple weeks. And definitely don't get your hopes up that I'll have good news. Nonetheless, I'm still going to try.

**Comment by stefansedich** [comment]
Thanks @2rs2ts appreciate the update, and I totally understand about missing notifications on old GitHub threads!

**Comment by bunnybilou** [comment]
Invested time last Friday attempting to configure the settings correctly, but encountered issues even with the latest version of the agent.

What's puzzling is that in the host page, the tag is only configured in the AWS section, not in the Datadog section. However, tags are still being set in metrics/logs.

![image](https://github.com/DataDog/datadog-agent/assets/10532612/5eb78091-480d-4e27-9ce1-a12ce3f7d04a) ![image](https://github.com/DataDog/datadog-agent/assets/10532612/c423eef0-d347-46fd-bbd5-9e9f0cf3e51c)

**Comment by 2rs2ts** [comment]
@bunnybilou if you haven't opened a ticket with datadog support, please do so and mention this github issue. It's a real pain and I'm getting frustrated that it's still not fixed, too.

**Comment by 2rs2ts** [comment]
I cannot understate how much I loathe bots like the above. Bugs/flaws don't go away just because nobody has actively talked about them for years. Most projects don't like it when you reply with +1 constantly on issues...

**Comment by chouetz** [comment]
Hi @2rs2ts,

Thank you for taking the time to share your feedback. We recently introduced a system to help manage long-standing issues through automated stale management. We understand this may have come as a surprise for some existing issues. We want to apologize if it felt that way. It was not our intent, and we want to reassure you that every issue is important to us and reviewed carefully. For your specific issue, we’re reviewing it to ensure the discussion progresses and receives the right level of attention. Thank you for helping improve the Datadog Agent.


— The Datadog Agent Team

**Comment by jszwedko** [comment]
Hey all!

Just to provide a bit more color here: `DD_COLLECT_EC2_TAGS` was originally implemented to have the Agent collect EC2 tags via IMDS if the AWS integration is not enabled (which also scrapes the tags). I can definitely understand the confusion, though, and we'll be looking into this a bit more to see if we should be coupling this collection so that if this option is disabled in the Agent it also disables collection in the AWS Integration.

> Same problem. We are trying to use dogstatsd but the custom metrics count just exploded because of this. (version 7.21.0 BTW)

Since these tags are correlated with the `host` tag their inclusion shouldn't increase the cardinality of the metrics over the `host` tag itself. Maybe I'm missing something though? Have people seen an increase in cardinality due to these tags and can share more details?

> I cannot understate how much I loathe bots like the above. Bugs/flaws don't go away just because nobody has actively talked about them for years. Most projects don't like it when you reply with +1 constantly on issues...

For sure, we are just starting to get an automated triage process in place but the intent is to label issues that have been validated and they won't require constant bumping to appease the stale bot. The primary intent is to close issues that are awaiting author response.

Also hi @2rs2ts 🙂

**Comment by gavinclarkeuk** [comment]
> Since these tags are correlated with the host tag their inclusion shouldn't increase the cardinality of the metrics over the host tag itself. Maybe I'm missing something though?

@jszwedko whilst you are probably right, I suspect that if any tags on the EC2 instance were to change over time then that would increase cardinality as presumably metrics would get enriched with the tags at that point in time. I haven't tested this.

For us, whilst cardinality is a concern, the issue is more the vast amount of clutter all those extra tags create in the Datadog UI. Whilst it might seem like a small thing it is slowing down engineers a tiny bit every day as they have to wade through a huge list trying to find the tags they care about.

It also makes it harder to manage the ingested tags for custom metrics, which impacts how easy it is to reduce metrics costs.

**Comment by jszwedko** [comment]
> > Since these tags are correlated with the host tag their inclusion shouldn't increase the cardinality of the metrics over the host tag itself. Maybe I'm missing something though?
> 
> [@jszwedko](https://github.com/jszwedko) whilst you are probably right, I suspect that if any tags on the EC2 instance were to change over time then that would increase cardinality as presumably metrics would get enriched with the tags at that point in time. I haven't tested this.

That's true, there is a short period of "tag churn" where the old tags are removed and the new tags added that can lead to temporary increases in cardinality.
 
> For us, whilst cardinality is a concern, the issue is more the vast amount of clutter all those extra tags create in the Datadog UI. Whilst it might seem like a small thing it is slowing down engineers a tiny bit every day as they have to wade through a huge list trying to find the tags they care about.
> 
> It also makes it harder to manage the ingested tags for custom metrics, which impacts how easy it is to reduce metrics cost

For sure, I can understand that: that the extraneous tags clutter the interface and make it difficult to highlight the tags that you actually care about.

**Comment by 2rs2ts** [comment]
> Thank you for taking the time to share your feedback. We recently introduced a system to help manage long-standing issues through automated stale management. We understand this may have come as a surprise for some existing issues. We want to apologize if it felt that way. It was not our intent, and we want to reassure you that every issue is important to us and reviewed carefully.

Appreciated, @chouetz. I get why people use bots like these. I just am very tired of my inbox being full of `/remove-lifecycle stale` for K8s issues, and many other such cases. It creates a maintenance burden for bug reports and feature requests that _in aggregate_ exceeds the burden and waste of human lifespan compared to closing issues that are resolved (but were not properly linked such that they would auto-resolve when a PR was merged or whatever.) It inflicts that burden of bumping stuff for years, and having to clear your inbox full of these automated messages and people's replies to them, both on the users and on yourselves. Even worse, it also leads to situations where completely valid extant issues get swept under the rug because someone chose the wrong time to go on vacation. It would be better to do nothing at all than to use auto-close bots.

> we'll be looking into this a bit more to see if we should be coupling this collection so that if this option is disabled in the Agent it also disables collection in the AWS Integration.

That would be excellent for things like Kubernetes. Thank you @jszwedko (also hello! It's been a while! 👋 )
